File: hw/fuentes/U3/HW20COMPOSITE/HW20COMPOSITE_PY/ec/edu/espe/enterprise/model/President.py
from ec.edu.espe.enterprise.model.Supervisor import Supervisor

class President(Supervisor):
    __instance = None

    def __init__(self, name="not assigned yet"):
        if President.__instance is not None:
            raise Exception("This class is a singleton!")
        super().__init__(name, "President")
        President.__instance = self

    @staticmethod
    def get_president(name):
        if President.__instance is None:
            President(name)
        President.__instance.name = name
        return President.__instance

    def state_name(self):
        super().state_name()

File: hw/astudillo/U3/Hw19Composite/Composite/src/main/java/ec/edu/espe/composite/composite/model/President.java
package ec.edu.espe.composite.composite.model;

public class President extends Supervisor {
    private static President president = new President();
    
    private President(String aName) {
        this();
        name = aName;
    }
    
    private President() {
        super();
        title = "President";
    }
    
    public void stateName() {
        super.stateName(title, name);
    }
    
    public static President getPresident(String aName) {
        president.name = aName;
        return president;
    }
}
